File: backend/dashboard-service/Backend.Dashboards.Api/Dto/IssueAssigneeAddedEvent.cs
using System.Text.Json.Serialization;

namespace Backend.Shared.DTOs;

public class IssueAssigneeAddedEvent
{
    [JsonPropertyName("projectId")]
    public long ProjectId { get; set; }

    [JsonPropertyName("issueId")]
    public long IssueId { get; set; }

    [JsonPropertyName("assignedUserId")]
    public long AssignedUserId { get; set; }

    [JsonPropertyName("assignerId")]
    public long AssignerId { get; set; }

    [JsonPropertyName("assignedAtUtc")]
    public DateTime AssignedAtUtc { get; set; } = DateTime.UtcNow;
}

File: backend/issue-service/src/main/java/com/example/issueservice/dto/rabbit/IssueAssigneeAddedEvent.java
package com.example.issueservice.dto.rabbit;

import com.example.issueservice.dto.models.Issue;

import java.time.Instant;

public record IssueAssigneeAddedEvent(
        long projectId,
        long issueId,
        long assignedUserId,
        long assignerId,
        Instant assignedAtUtc
) {
    public static IssueAssigneeAddedEvent from(Issue issue, long assignedUserId, long assignerId) {
        return new IssueAssigneeAddedEvent(
                issue.getProjectId(),
                issue.getId(),
                assignedUserId,
                assignerId,
                Instant.now()
        );
    }
}
